lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx

```diff
@@ -302,11 +302,26 @@ export default function AssignmentActivity() {
           />
         )}
         {isAutoGradingAssignment && auto_grading_sync_enabled && (
-          <SyncGradesButton
-            studentsToSync={studentsToSync}
-            lastSync={lastSync}
-            onSyncScheduled={onSyncScheduled}
-          />
+          <>
+            {!lastSync.canceled ? (
+              <SyncGradesButton
+                studentsToSync={studentsToSync}
+                lastSync={lastSync}
+                onSyncScheduled={onSyncScheduled}
+              />
+            ) : (
+              <div
+                className={classnames(
+                  'rounded p-2',
+                  'border border-grade-error',
+                  'font-bold text-grade-error bg-grade-error-light',
+                )}
+                data-testid="long-sync-error"
+              >
+                Syncing is taking too long. Reload the page.
```

Comment:
Just remembered we discuss suggesting the user to open a support issue, but maybe for that the default time should be a bit higher.

Reply:
30 seconds does seem like it could be too short if there are hundreds of grades to sync. @marcospri do you have any thoughts on what a reasonable upper bound would be on how long the frontend should poll before it should time out. For context, we have a timeout to prevent the frontend from polling the backend indefinitely if syncing gets stuck for any reason.

Reply:
I don't have it yet, we have know a course to test this with 25 students but I'd like to finish some things before that.

An API call to the LMS will timeout after 30 seconds. We'll retry it up to two times with a 10s backoff. So the maximum wait for individual grades should be less than 2 minutes.  The global sync might take a bit longer to gather the results.


I'd err on the side of being to long and reduce it later if excessive. Maybe 2 minute to start with? 

Two minutes is a long time but if there are a lot of students you'll  get feedback for along the way as some task succeed/fail.